wp_mobile_cache/src/repository/term_relationships.rs

```diff
@@ -0,0 +1,483 @@
+            placeholders.join(", ")
+        );
+
+        // Build params: [site_id, object_id, taxonomy_type, term_id1, term_id2, ...]
+        let mut params: Vec<Box<dyn rusqlite::ToSql>> = vec![
+            Box::new(site.row_id),
+            Box::new(object_id),
+            Box::new(taxonomy_type.to_string()),
+        ];
+        for term_id in term_ids {
+            params.push(Box::new(term_id.0));
+        }
+
+        let params_refs: Vec<_> = params.iter().map(|p| p.as_ref()).collect();
+        executor.execute(&sql, params_refs.as_slice())?;
+        Ok(())
+    }
+
+    /// Insert new terms for an object.
+    fn insert_terms(
+        &self,
+        executor: &impl QueryExecutor,
+        site: &DbSite,
+        object_id: RowId,
+        taxonomy_type: &TaxonomyType,
+        term_ids: &[TermId],
+    ) -> Result<(), SqliteDbError> {
+        if term_ids.is_empty() {
+            return Ok(());
+        }
+
+        for term_id in term_ids {
+            executor.execute(
+                "INSERT INTO term_relationships (db_site_id, object_id, term_id, taxonomy_type) VALUES (?, ?, ?, ?)",
+                rusqlite::params![site.row_id, object_id, term_id.0, taxonomy_type.to_string()],
+            )?;
+        }
+        Ok(())
+    }
+
+    /// Get all term IDs for an object's taxonomy.
+    pub fn get_terms_for_object(
+        &self,
+        executor: &impl QueryExecutor,
+        site: &DbSite,
+        object_id: RowId,
+        taxonomy_type: &TaxonomyType,
+    ) -> Result<Vec<TermId>, SqliteDbError> {
+        let sql = "SELECT term_id FROM term_relationships WHERE db_site_id = ? AND object_id = ? AND taxonomy_type = ?";
+        let mut stmt = executor.prepare(sql)?;
+        let rows = stmt.query_map(
+            rusqlite::params![site.row_id, object_id, taxonomy_type.to_string()],
+            |row| {
+                let id: i64 = row.get(0)?;
+                Ok(TermId(id))
+            },
+        )?;
+
+        rows.collect::<Result<Vec<_>, _>>()
+            .map_err(SqliteDbError::from)
+    }
+
+    /// Get all term IDs grouped by taxonomy for an object (for post reads with joins).
+    pub fn get_all_terms_for_object(
+        &self,
+        executor: &impl QueryExecutor,
+        site: &DbSite,
+        object_id: RowId,
+    ) -> Result<HashMap<TaxonomyType, Vec<TermId>>, SqliteDbError> {
+        let sql = "SELECT taxonomy_type, term_id FROM term_relationships WHERE db_site_id = ? AND object_id = ?";
+        let mut stmt = executor.prepare(sql)?;
+        let rows = stmt.query_map(rusqlite::params![site.row_id, object_id], |row| {
+            let taxonomy_str: String = row.get(0)?;
+            let term_id: i64 = row.get(1)?;
+            Ok((taxonomy_str, term_id))
+        })?;
+
+        let mut result: HashMap<TaxonomyType, Vec<TermId>> = HashMap::new();
+        for row_result in rows {
```

Comment:
Would be good to use functional programming if we can. (there are other for statements in this PR)